src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -0,0 +1,319 @@
+    String recordString = null;
+    ContentItem contentItem = null;
+
+    if (!config.isEntireRecord()) {
+      recordString = record.get(config.getFieldName()).toString();
+    } else {
+      recordString = StructuredRecordStringConverter.toDelimitedString(record, ",");
+    }
+    try {
+
+      // depending on input schema field object
+      contentItem = ContentItem.newBuilder().setValue(recordString).build();
+
+      String templateName = String.format("projects/%s/inspectTemplates/%s", config.getProject(), config.templateId);
+
+      InspectContentRequest request =
+        InspectContentRequest.newBuilder()
+                             .setParent(ProjectName.of(config.getProject()).toString())
+                             .setInspectTemplateName(templateName)
+                             .setItem(contentItem)
+                             .build();
+
+      metrics.count("dlp.requests.count", 1);
+      InspectContentResponse response = client.inspectContent(request);
+      InspectResult result = response.getResult();
+
+      metrics.count("dlp.requests.success", 1);
+      if (result.getFindingsList().size() > 0) {
+        emitter.emit(SENSITIVE_PORT, record);
+        return;
+      }
+
+      emitter.emit(NON_SENSITIVE_PORT, record);
+
+    } catch (Exception e) {
+      metrics.count("dlp.requests.fail", 1);
+      if (e instanceof ResourceExhaustedException) {
+        LOG.error(
+          "Failed due to DLP rate limit, please request more quota from DLP: https://cloud.google"
+            + ".com/dlp/limits#increases");
+        throw e;
+      }
+
+      switch (config.onErrorHandling()) {
+        case -1:
+          throw new Exception("Terminating pipeline on error as set in plugin configuration." + e.getMessage());
+        case 0:
+          return;
+        case 1:
+          emitter.emitError(new InvalidEntry<>(-1, e.getMessage(), record));
+          break;
+      }
+
+    }
+  }
+
+  @Override
+  public void destroy() {
+    super.destroy();
+    if (client != null) {
+      client.close();
+    }
+  }
+
+
+  /**
+   * Configures the <code>DlpSettings</code> to use user specified service account file or auto-detect.
+   *
+   * @return Instance of <code>DlpServiceSettings</code>
+   * @throws IOException thrown when there is issue reading service account file.
+   */
+  private DlpServiceSettings getSettings() throws IOException {
+    DlpServiceSettings.Builder builder = DlpServiceSettings.newBuilder();
+    if (config.getServiceAccountFilePath() != null) {
+      builder.setCredentialsProvider(() -> GCPUtils.loadServiceAccountCredentials(config.getServiceAccountFilePath()));
+    }
+    return builder.build();
+  }
+
+
+  /**
+   * Configuration object.
+   */
+  public static class Config extends GCPConfig {
+
+    @Macro
+    @Name("entire-record")
+    @Description("Check full record or a field")
+    private String entireRecord;
```

Comment:
can this be just a boolean? Would it ever hold information other than record or field?

Reply:
Made it a boolean